From 2f3054161cdd6f1c7a3f205a60e538b7f022f602 Mon Sep 17 00:00:00 2001
From: wei li <sirius.liwei@honor.corp-partner.google.com>
Date: Fri, 5 Dec 2025 15:48:26 +0800
Subject: [PATCH 04/15] ANDROID: ABI: update symbol list for honor

Add the symbols used by CIFS.

3 function symbol(s) added
  'void d_delete(struct dentry*)'
  'int filemap_fdatawrite_range_kick(struct address_space*, loff_t, loff_t)'
  'void sk_net_refcnt_upgrade(struct sock*)'

Bug: 466263108
Bug: 466897784
Change-Id: Ic5d29070e55cae03475a540e4a16e6d03bfdd9ac
Signed-off-by: wei li <sirius.liwei@honor.corp-partner.google.com>
(cherry picked from commit 6603bb3a8e4dc5131ece4170f62162ac25abf517)
---
 gki/aarch64/abi.stg       | 30 ++++++++++++++++++++++++++++++
 gki/aarch64/symbols/honor |  3 +++
 2 files changed, 33 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 4bc8a2801b02..7b8b864d6a35 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -456361,6 +456361,15 @@ elf_symbol {
   type_id: 0x880a3e61
   full_name: "d_alloc_parallel"
 }
+elf_symbol {
+  id: 0x2023f91e
+  name: "d_delete"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xdc69387f
+  type_id: 0x141901a7
+  full_name: "d_delete"
+}
 elf_symbol {
   id: 0xfeea461b
   name: "d_drop"
@@ -468522,6 +468531,15 @@ elf_symbol {
   type_id: 0x98773f07
   full_name: "filemap_fdatawrite_range"
 }
+elf_symbol {
+  id: 0xa62cfc91
+  name: "filemap_fdatawrite_range_kick"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xaa7dca80
+  type_id: 0x98773f07
+  full_name: "filemap_fdatawrite_range_kick"
+}
 elf_symbol {
   id: 0x7c937fc2
   name: "filemap_flush"
@@ -499905,6 +499923,15 @@ elf_symbol {
   type_id: 0x9a667a13
   full_name: "sk_msg_zerocopy_from_iter"
 }
+elf_symbol {
+  id: 0x216a640a
+  name: "sk_net_refcnt_upgrade"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x49300540
+  type_id: 0x17c95d08
+  full_name: "sk_net_refcnt_upgrade"
+}
 elf_symbol {
   id: 0xb00a73bc
   name: "sk_psock_drop"
@@ -520147,6 +520174,7 @@ interface {
   symbol_id: 0x171eca8a
   symbol_id: 0x82537a82
   symbol_id: 0x4b8dc0ec
+  symbol_id: 0x2023f91e
   symbol_id: 0xfeea461b
   symbol_id: 0x9534fec7
   symbol_id: 0x22c4092b
@@ -521495,6 +521523,7 @@ interface {
   symbol_id: 0xef606c3f
   symbol_id: 0x3289d59f
   symbol_id: 0x82b19820
+  symbol_id: 0xa62cfc91
   symbol_id: 0x7c937fc2
   symbol_id: 0x667b7dc2
   symbol_id: 0xb11f578f
@@ -524979,6 +525008,7 @@ interface {
   symbol_id: 0x154f6c3d
   symbol_id: 0x8da4646f
   symbol_id: 0x1befb3d5
+  symbol_id: 0x216a640a
   symbol_id: 0xb00a73bc
   symbol_id: 0x31b3c589
   symbol_id: 0xe6c00408
diff --git a/gki/aarch64/symbols/honor b/gki/aarch64/symbols/honor
index 1a78cf519113..a4b74f5ab8ef 100644
--- a/gki/aarch64/symbols/honor
+++ b/gki/aarch64/symbols/honor
@@ -584,4 +584,7 @@
   __traceiter_android_vh_append_total_power
   __traceiter_android_rvh_dpm_prepare
   __tracepoint_android_rvh_dpm_prepare
+  sk_net_refcnt_upgrade
+  d_delete
+  filemap_fdatawrite_range_kick
 
-- 
2.50.1

